Formula/coreutils.rb

```diff
@@ -49,16 +51,29 @@ def install
     args = %W[
       --prefix=#{prefix}
       --program-prefix=g
-      --without-gmp
+      --with-gmp
       --without-selinux
     ]
 
     system "./configure", *args
     system "make", "install"
 
+    no_conflict = if OS.mac?
+      []
+    else
+      %w[
+        b2sum base32 basenc chcon dir dircolors factor hostid md5sum nproc numfmt pinky ptx realpath runcon
+        sha1sum sha224sum sha256sum sha384sum sha512sum shred shuf stdbuf tac timeout truncate vdir
+      ]
+    end
+
     # Symlink all commands into libexec/gnubin without the 'g' prefix
     coreutils_filenames(bin).each do |cmd|
       (libexec/"gnubin").install_symlink bin/"g#{cmd}" => cmd
+
+      # Find non-conflicting commands on macOS
+      which_cmd = which(cmd)
+      no_conflict << cmd if OS.mac? && (which_cmd.nil? || !which_cmd.to_s.start_with?(%r{(/usr)?/s?bin}))
```

Comment:
This will cause a regression: https://github.com/Homebrew/homebrew-core/commit/1b198c839ba376943aa51988ba0a84570214e121